### PR TITLE
Add file extend plugin

### PR DESCRIPTION
**Pull request by benurb**
Hey guys,

I added a file extend plugin as described in #6 . Let me know what you think? 👍 

Ben

### CODE REVIEW

File: plugins/extend/file/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"use strict";
+
+var plugin = require("alamid-plugin"),
+    merge = require("../merge.js"),
```
Comment (meaku) [inline plugins/extend/file/index.js]: Probably just copy&paste as you don't use it anywhere.

File: plugins/extend/file/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    fs = require("fs"),
+    path = require("path"),
+    deepAssign = require("deep-assign");
+
+/**
+ * merge from env into config
+ * @type {Function|*|exports}
+ */
+var filePlugin = plugin(function (obj, fileExtension, whitelist) {
```
Comment (meaku) [inline plugins/extend/file/index.js]: I guess we don't need a whitelist in this case?

**Comment by benurb** [comment]
Hey @meaku,

I updated the PR and fixed your remarks. Looks good? 😄 

Ben

**Comment by jhnns** [comment]
LGTM. I'll need to do some refactoring though because it does not comply to our coding rules 👅 (no `var` inside `if`).

@meaku Regarding options in plugins: We should a) document them or b) remove them. In any case, we should use `options` object instead of multiple arguments because it's always a PITA to pass options as separate arguments.

**Comment by jhnns** [comment]
Shipped with 0.3.0

**Pull request merged**